routemaster/models/batch.rb

```diff
@@ -26,10 +26,11 @@ class Batch
       attr_reader :uid, :deadline
 
 
-      def initialize(uid:, deadline: nil, subscriber: nil)
+      def initialize(uid:, deadline: nil, subscriber: nil, length: nil)
         @uid        = uid
         @deadline   = deadline
         @subscriber = subscriber
+        @_length    = length
```

Comment:
It is used, just below - line 50 in this file. The purpose of this extra argument is to prepopulate the `@_length` memoisation ivar for the `#length` method.

The reason is that in common instanciation patterns of a `Batch` we'll actually know what the length is, so this saves a roundtrip to Redis. Given the low-latency nature of this work... even roundtrip saved helps.

Does this make sense?